src/commands/rollout-image.yml

```diff
@@ -16,10 +16,17 @@ parameters:
     description: Docker image tag
     type: string
     default: latest
+  namespace:
+    description: "The Kubernetes namespace name."
+    type: string
+    default: ""
 steps:
   - update-kubeconfig-with-credentials:
       cluster: <<parameters.cluster>>
   - run:
       name: Roll out image
       command: |
-        kubectl set image deployment <<parameters.deployment>> <<parameters.container>>=<<parameters.image>>:<<parameters.tag>>
+        if [ -n "<<parameters.namespace>>" ]; then
+          set -- "$@" --namespace="<<parameters.namespace>>"
```

Comment:
`$@` is suppose to list all arguments passed to Bash. It seems counter-intuitive to set it manually here vs using a dedicated envar.

I would think a line like this would be better:

```
namespaceFlag='--namespace="<<parameters.namespace>>"'
```

Reply:
@felicianotech I felt that this would cater well for additional parameters in the future, what do you think?

Reply:
How so?

Reply:
@felicianotech If there are other parameters to be supported in the future, similar conditional blocks can be added to build `$@` up. And we would not have to edit the `kubectl set image deployment` line. It feels like a clean way to cater to different combinations of optional parameters.

Reply:
I see. I still thing the variable should have a different name though since `$@` already has a meaning. Maybe something like `optionalParams` or whatever.

Reply:
@felicianotech I'm actually using the name `$@` by choice, so that I can have an array in a POSIX compliant way, since there isn't an easy way to do arrays in POSIX shells, but the positional parameter array `$@` is a sort of built-in array available in POSIX shells, if I am not mistaken. I think this StackOverflow answer (https://stackoverflow.com/questions/35385962/arrays-in-a-posix-compliant-shell/59353917#59353917) and the "Working with arrays" section on this page http://www.etalabs.net/sh_tricks.html are quite representative of what I'm trying to do with `$@` and `set`.

Reply:
I'm pretty sure we agreed as a team that we we're only going to bother supporting Bash. I would do this a different way but clearly this is a also a common practice in the wild. So looks fine to me.